libethereum/Client.cpp

```diff
@@ -226,6 +226,8 @@ void Client::init( WithExisting _forceAction, u256 _networkId ) {
     if ( m_dbPath.size() )
         Defaults::setDBPath( m_dbPath );
 
+    bc().m_stateDB = m_state.db();
+
```

Comment:
Why was this necessary?

Reply:
This is initialization of variable which must be non-null smart pointer earlier

Reply:
I meant this:
`bc().m_stateDB = m_state.db();`

This looks like intruding into BlockChain class' private parts..

Reply:
yes, right, but I need it initialized